.github/workflows/main.yml

```diff
@@ -0,0 +1,253 @@
+
+      - name: Publish AdminConsole
+        run: >
+          dotnet publish src/AdminConsole/
+          --no-restore
+          --no-build
+          --configuration Release
+          --runtime win-x86
+          --no-self-contained
+          --output publish/AdminConsole/
+          -p:ContinuousIntegrationBuild=true
+          ${{ steps.prerelease-version.outputs.version && format('-p:Version={0}', steps.prerelease-version.outputs.version) || '' }}
+
+      - name: Upload API artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: Api
+          path: publish/Api/
+
+      - name: Upload AdminConsole artifacts
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          name: AdminConsole
+          path: publish/AdminConsole/
+
+  # Deploy the apps
+  deploy:
+    needs:
+      # Technically, it's not required for the format job to succeed for us to push the package,
+      # so we may consider removing it as a prerequisite here.
+      - format
+      - test
+      - pack
+    runs-on: ubuntu-latest
+    environment: devtest
+    permissions:
+      actions: read
+      packages: write
+
+    steps:
+      - name: Download API artifacts
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: Api
+
+      - name: Download AdminConsole artifacts
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: AdminConsole
+
+      - name: Archive API
+        run: zip -r Api.zip publish/Api/
+
+      - name: Archive AdminConsole
+        run: zip -r AdminConsole.zip publish/AdminConsole/
+
+      - name: Login to Azure
+        uses: Azure/login@1f63701bf3e6892515f1b7ce2d2bf1708b46beaf  # v1.4.3
+        with:
+          creds: ${{ secrets.APP_SERVICE_SP }}
+
+      - name: Configure app settings
+        # Temporary
+        if: ${{ false }}
+        run: >
+          az webapp config appsettings set
+          --resource-group ${{ secrets.AZURE_RESOURCE_GROUP_NAME }}
+          --name ${{ secrets.AZURE_APP_NAME }}
+          --slot staging
+          > /dev/null
+
+      - name: Push API
+        # Temporary
+        if: ${{ false }}
+        run: >
+          az webapp deployment source config-zip
+          --resource-group ${{ secrets.AZURE_RESOURCE_GROUP_NAME }}
+          --name ${{ secrets.AZURE_APP_NAME }}
+          --slot staging
+          --src Api.zip
+
+      - name: Push AdminConsole
+        # Temporary
+        if: ${{ false }}
+        run: >
+          az webapp deployment source config-zip
+          --resource-group ${{ secrets.AZURE_RESOURCE_GROUP_NAME }}
+          --name ${{ secrets.AZURE_APP_NAME }}
+          --slot staging
+          --src AdminConsole.zip
```

Comment:
This currently deploys to staging, but we can also deploy to production. If we're worried about leaking secrets, we can have another job in a private repo and run it here via a secret webhook (that way it's still triggered by this workflow).

Reply:
I think another repo is a good approach for now. I can set one up.